### ISSUE TITLE
Create fourth status for unknown delivery status

### ISSUE THREAD

**Issue by yaelberger-commits** [cds-snc/notification-planning#1138]
## Create fourth status for unknown delivery status

As a Sender, I need to be able to understand when a notification I sent has an unknown status and will not be retried so that I can report on the status of my messages, like if I send to an international phone number and that country doesn't provide delivery receipts to AWS.

As the Notify team, we need to be able to report unknown status on a users service dashboard so that users can better understand that some messages never get confirmation and won't ask us that question through support.

WHY are we building?
We get questions from Senders through support when they are confused by notifications that remain stuck in transit and never land in either the delivered or failed status.

WHAT are we building?
A new status column in the db and dashboard for unknown delivery status

VALUE created by our solution
More clarity and less confusion for Senders, and a reduction in support tickets about delivery status.
Less time spent for Ops Lead investigating messages that appear stuck in transit

Example support ticket https://cds-snc.freshdesk.com/a/tickets/20119

## Acceptance Criteria
Given a notification was sent by Notify and AWS but never received back a delivery receipt, after X amount of time, then the notification will appear in the "unknown" status in the db and the service dashboard.

- [ ] Generate appropriate log messages so that executions of this feature can be tracked
- [ ] Can misuse of this feature cause harm? If yes, create an alert
- [ ] Update the status of related findings, insights, and hypotheses on the Research Airtable
- [ ] Once change/fix/feature is implemented, link relevant Airtable records to design artifacts (Figma)

* A11y
* Bilingualism
* Privacy considerations
* Security controls in place
* Measuring success and metrics

## Related [Research Airtable](https://airtable.com/appWwAmHwDLtpIyko/tbl38n7ofWYBuezFc/viwQUX3tUxbMxYS7C?blocks=hide) records
- (Ping Adrianne to add stuff here or find stuff yourself)

## QA Steps
- [ ] Tested in a realistic production scenario

**Comment by yaelberger-commits** [comment]
@whabanks this task is partially completed by your work on international numbers, correct? Do you think we still need a forth status for unknown?

**Comment by jzbahrai** [comment]
@yaelberger-commits you are right, this ticket is mainly completed by #1760. Going to shut this as we added a ProviderError for delivery status explanations.

**Comment by yaelberger-commits** [comment]
Bringing this back into the product backlog again as we still have issues with email notifications stuck in sending for a day. See thread here: https://gcdigital.slack.com/archives/C03FA4DJCCU/p1747330258445909
